ciao-controller/main.go

```diff
@@ -283,3 +295,33 @@ func (c *controller) startCiaoService() error {
 
 	return nil
 }
+
+func getControllerFQDNFromCert(cert string) (string, error) {
+	bytesCert, err := ioutil.ReadFile(cert)
+	if err != nil {
+		return "", fmt.Errorf("couldn't read cert: %s", err)
+	}
+
+	blockCert, _ := pem.Decode(bytesCert)
+	if blockCert == nil {
+		return "", fmt.Errorf("couldn't decode cert")
+	}
+
+	parsedCert, err := x509.ParseCertificate(blockCert.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("couldn't parse cert: %s", err)
+	}
+
+	role := ssntp.GetRoleFromOIDs(parsedCert.UnknownExtKeyUsage)
+	if !role.IsController() {
+		return "", fmt.Errorf("expected certificate role ssntp.Controller, got role %s", role.String())
```

Comment:
I'm not super convinced that pulling it from the SSNTP cert is a good idea. The SSNTP communication could be on it's own private network with "internal" dns entries like cn01.ciao.internal, controller.ciao.internal, scheduler.ciao.internal and whereas controller's HTTP service would be on a more exposed network and name.

If we can't extract the hostname from the HTTP cert then i'm leaning towards:

1. Keeping the configuration entry or..
2. Removing any api where we need to expose absolute URIs to the client